pkg/plugin/restore_pvc_action_plugin.go

```diff
@@ -24,25 +31,66 @@ func (p *NewPVCRestoreItemAction) AppliesTo() (velero.ResourceSelector, error) {
 }
 
 func (p *NewPVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
-	var pvc corev1api.PersistentVolumeClaim
+	var pvc corev1.PersistentVolumeClaim
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(input.Item.UnstructuredContent(), &pvc); err != nil {
 		return nil, errors.WithStack(err)
 	}
 
-	p.Log.Infof("VSphere PVCRestoreItemAction for PVC %s/%s started", pvc.Namespace, pvc.Name)
 	var err error
+	// get snapshot blob from PVC annotation and reset PVC annotation
+	snapshotAnnotation, ok := pvc.Annotations[utils.ItemSnapshotLabel]
+	if !ok {
+		p.Log.Infof("Skipping PVCRestoreItemAction for PVC %s/%s, PVC does not have a vSphere BackupItemAction snapshot.", pvc.Namespace, pvc.Name)
+		return &velero.RestoreItemActionExecuteOutput{
+			UpdatedItem: input.Item,
+		}, nil
+	}
+	delete(pvc.Annotations, utils.ItemSnapshotLabel)
```

Comment:
Again, as discussed offline, do not delete this snap info. I will be using this during the DeleteItemAction

Reply:
I double checked this case, and thought it doesn't matter whether we delete the annotation here or not. 

For both `velero restore` and `velero backup delete`, the unstructured item(PVC in our case) comes from object store rather than the local k8s cluster. As long as we don't overwrite/remove the annotation of PVC object in object store, we are good.

Reply:
So does the object store have the PVC persisted at the point in time when backup was created?

Reply:
Yes, PVC object with our annotation will be persisted when velero backup is persisted.